### PR TITLE
Return GeoDataFrame in the to_dataframe function

### PR DESCRIPTION
**Pull request by antoniocarlon**
- Return `GeoDataFrame` in the `to_dataframe` function
 - Modified docstring
 - Added check to the tests (`assert isinstance(df, GeoDataFrame)`)

### CODE REVIEW

File: cartoframes/data/observatory/catalog/entity.py [hunk 1]
```diff
@@ -142,7 +144,12 @@ def _download(self, credentials, file_path=None, limit=None, order_by=None, sql_
                 log.info(_GEOGRAPHY_READ_MSG.format(file_path))
         else:
             dataframe = pd.read_csv(rows)
-            return dataframe
+            gdf = GeoDataFrame(dataframe)
+
+            if has_geometry(dataframe):
```
Comment (Jesus89) [inline cartoframes/data/observatory/catalog/entity.py:149]: The function `has_geometry` check the geometry of an existing GeoDataFrame, not for a DataFrames, so this won't work. We should implement the following logic instead:
- Check if the existing dataframe contains a raw geom column if:
  - The entity is a Geography or
  - The entity is a Dataset and add_geom is True
- In every case check that the column is there and can be decoded:
  - decode_geometry(df['geom'])
- If it works, create the GeoDataFrame with that geometry
- PD: we need to decide if we should keep the original geom column or not

**Review by Jesus89 — CHANGES_REQUESTED** [review]
Good approach. However, it requires some changes into the logic, and to cover these changes with unit tests.

File: tests/e2e/data/observatory/catalog/test_download.py [hunk 2]
```diff
@@ -75,6 +75,7 @@ def test_dataset_to_csv_private(self):
     def test_dataset_to_dataframe_public(self):
         df = public_dataset.to_dataframe(self.credentials, limit=PUBLIC_LIMIT)
         assert isinstance(df, GeoDataFrame)
+        assert df.head().geom.type == 'Polygon'
```
Comment (Jesus89) [inline tests/e2e/data/observatory/catalog/test_download.py:78]: I think you could access directly the column `assert df.geom.type == 'Polygon'`

Comment (antoniocarlon) [reply tests/e2e/data/observatory/catalog/test_download.py]: Yes, you are right, changing it :+1:

**Review by Jesus89 — APPROVED** [review]

**Pull request merged**